logging/lazy_fields.go

```diff
@@ -0,0 +1,189 @@
+package logging
+
+import (
+	"encoding/hex"
+	"fmt"
+	spec "github.com/attestantio/go-eth2-client/spec/phase0"
+	specqbft "github.com/bloxapp/ssv-spec/qbft"
+	spectypes "github.com/bloxapp/ssv-spec/types"
+	"github.com/bloxapp/ssv/protocol/v2/blockchain/beacon"
+	protocolp2p "github.com/bloxapp/ssv/protocol/v2/p2p"
+	"github.com/dgraph-io/ristretto"
+	"math/big"
+	"net"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+
+	"github.com/bloxapp/ssv/network/records"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var (
+	messageNotLogged = "heavy field not logged"
+	showHeavyFields  = false
+)
+
+func init() {
+	if os.Getenv("SHOW_HEAVY_FIELDS") == "true" {
+		showHeavyFields = true
+	}
+
+	//other checks, f.e. command line args or config file
+}
+
+func LazyBigInt(key string, val *big.Int) zapcore.Field {
```

Comment:
please write simple unit tests for all new functions

Reply:
for LazyBigInt the behavior was 
`zap.Uint64("syncOffset", syncOffset.Uint64()))`
but now you are using .String(). Is your implementation correct?